diagnostics/mem.go

```diff
@@ -0,0 +1,38 @@
+package diagnostics
+
+import (
+	"encoding/json"
+	"github.com/ledgerwatch/erigon-lib/common/dbg"
+	"net/http"
+	"time"
+)
+
+func SetupMemAccess(metricsMux *http.ServeMux) {
+	metricsMux.HandleFunc("/mem", func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Access-Control-Allow-Origin", "*")
+		w.Header().Set("Content-Type", "application/json")
+		writeMem(w)
+	})
+
+	// update prometheus memory stats at least every 30 seconds
+	go func() {
+		ticker := time.NewTicker(30 * time.Second)
+		defer ticker.Stop()
+
+		for {
```

Comment:
this is an infinite loop which will never end and keep the goroutine in use forever - that will cause unclean process exits (the process will be stuck when exiting and it would force users to use kill -9)

a way to solve this is to pass in a `context.Context` object which will get cancelled on program exit and use it in a select like this:

```
select {
case <-ctx.Done():
    return
case <-ticker.C:
    // do stuff
```

can read more about select online